Домашние здания Седухин Петр

petr-sed opened the pull request.

petr-sed committed:
Выполнил легуий уровень ДЗ первого урока

petr-sed committed:
Первая часть готова

petr-sed committed:
Сделал второе задание

petr-sed committed:
сделал третью часть

petr-sed committed:
создал меню выбора

petr-sed committed:
Добавил обработку дробей в функцию inp_var

petr-sed committed:
добавил обработку исключений TypeError и ValueError в функцию inp_var

petr-sed committed:
В процессе создания верификации

manchenkoff commented on lesson01/home_work/hw01_normal.py:
```diff
@@ -1,27 +1,102 @@
+def inp_var(intro_long, intro_short, int_float_str):
+    try:
+        if intro_long == 0:
+            vr = int_float_str(input('Укажите значение переменной ' + intro_short + ': '))
```
функции типа int_float_str довольно небезопасны, лучше ограничиться явными int или float и тд

petr-sed committed:
выполнил первое задание

petr-sed committed:
выполнил easy второе задание

petr-sed committed:
выполнил easy

petr-sed committed:
Первое задание normal

petr-sed committed:
второе normal

petr-sed committed:
третье normal

petr-sed committed:
закончил normal

petr-sed committed:
первую часть hard решил как понял, вторую решил

petr-sed committed:
Первое задание easy

petr-sed committed:
Вторая часть easy

petr-sed committed:
Первая часть normal

petr-sed committed:
втоая часть normal

petr-sed committed:
третья часть normal

petr-sed committed:
четвертая часть normal

petr-sed committed:
мелкие правки

petr-sed committed:
задание 3 hard

petr-sed committed:
задание 2 hard

petr-sed committed:
easy 1

petr-sed committed:
easy 2

petr-sed committed:
easy 3

petr-sed committed:
normal 1

petr-sed committed:
Easy 1

petr-sed committed:
добавил dir path

petr-sed committed:
easy 2

petr-sed committed:
easy 3

petr-sed committed:
Создал функцию создания и удаления папок

petr-sed committed:
функция, возвращающая список папок

petr-sed committed:
убрал лишнее объявление content в третьей части

petr-sed committed:
добавил путь к третьей части

petr-sed committed:
закомментил все действия в easy для импорта. Выполнил normal.

petr-sed committed:
использовал конструкцию __name__ == __main__

petr-sed committed:
убрал  лишние возвраты к началу прграммы

petr-sed committed:
Добавил сообщение при переходе в папку в пункте 1

petr-sed committed:
Создал класс треугольник
__init__ c тремя точками
orient определяет правильную ориентацию фигуры на плоскости
sides рассчитывает длины сторон

petr-sed committed:
Написал функции  рассчета периметра, высоты и площади

petr-sed committed:
упростил округление

petr-sed committed:
класс трапеция
ориентация на плоскости
длины сторон
верификация соответствия равнобочной трапеции

petr-sed committed:
отдельная функция для вывода длин сторон
периметр
площадь

petr-sed committed:
в процессе

petr-sed committed:
затуп

petr-sed committed:
написал общую структуру

petr-sed committed:
разобрался с записью

petr-sed committed:
уверен, что можно было все сделать проще.

petr-sed committed:
научился искать линию с id в файле по name

petr-sed committed:
получаю id для запроса

petr-sed committed:
получил тестовую страницу с сайта погоды

petr-sed committed:
bd

petr-sed committed:
функционально описал процесс загрузки с сайта и записи в бд

petr-sed committed:
нашел apiid

petr-sed committed:
лишние бд

petr-sed committed:
файл бд

petr-sed committed:
Создал классы

petr-sed committed:
программа может:
1. Создавать файл базы данных SQLite
2. Выводить список стран из файла и список городов страны
3. Брать информацию о погоде в выбранном городе
4. Добавлять данные о погоде в базу

petr-sed committed:
Программа может работать со списком городов

petr-sed committed:
добавил элемент валидации для ввода
